patches/tModLoader/Terraria/ModLoader/ModExtraJumpLoader.cs

```diff
@@ -0,0 +1,218 @@
+				float runAccelerationMult = 1f;
+				float maxRunSpeedMult = 1f;
+				jump.HorizontalJumpSpeed(ref runAccelerationMult, ref maxRunSpeedMult);
+				player.runAcceleration *= runAccelerationMult;
+				player.maxRunSpeed *= maxRunSpeedMult;
+			}
+		}
+
+		/// <summary>
+		/// Do the jump if a valid <see cref="Player.activeJump"/> is found in <see cref="SetNextJump"/>.
+		/// </summary>
+		/// <param name="player">Jumping player</param>
+		internal static void Jump(Player player) {
+			bool playSound = true;
+			float jumpHeight = 1f;
+			ModExtraJump jump = player.activeJump;
+			jump.IsPerformingJump = true;
+			jump.Jump(ref jumpHeight, ref playSound);
+			if (playSound) {
+				SoundEngine.PlaySound(SoundID.DoubleJump, (int)player.position.X, (int)player.position.Y);
+			}
+			player.velocity.Y = -Player.jumpSpeed * player.gravDir;
+			player.jump = (int)(Player.jumpHeight * jumpHeight);
+		}
+
+		/// <summary>
+		/// Resets all hasJumpOption toggles.
+		/// </summary>
+		/// <param name="player">Jumping player</param>
+		internal static void ResetAllEnable(Player player) {
+			foreach (ModExtraJump extraJump in player.modExtraJumps.Where(j => !(j is VanillaExtraJump))) {
+				extraJump.hasJumpOption = false;
+			}
+		}
+
+		/// <summary>
+		/// Stops the ability to do modded extra jumps (finished jumping, on a rope, mounted, CCed or grappled).
+		/// </summary>
+		/// <param name="player">Jumping player</param>
+		internal static void ResetAllJumps(Player player) {
+			player.activeJump = null;
+			foreach (ModExtraJump extraJump in player.modExtraJumps.Where(j => !(j is VanillaExtraJump))) {
+				extraJump.canJumpAgain = false;
+			}
+		}
+
+		/// <summary>
+		/// Resets the jumping effect for all modded extra jumps.
+		/// </summary>
+		/// <param name="player">Jumping player</param>
+		internal static void ResetAllVisualEffects(Player player) {
+			foreach (ModExtraJump extraJump in player.modExtraJumps.Where(j => !(j is VanillaExtraJump))) {
+				extraJump.IsPerformingJump = false;
+			}
+		}
+
+		/// <summary>
+		/// Returns true if atleast one double jump is able to be used.
+		/// </summary>
+		/// <param name="player">Jumping player</param>
+		public static bool AnyJumpAgain(Player player) => player.modExtraJumps.Any(j => !(j is VanillaExtraJump) && j.canJumpAgain);
+
+		/// <summary>
+		/// Allows the ability to double jump if it is enabled.
+		/// </summary>
+		/// <param name="player">Jumping player</param>
+		internal static void SetJumpState(Player player) {
+			foreach (ModExtraJump extraJump in player.modExtraJumps.Where(j => !(j is VanillaExtraJump))) {
+				if (extraJump.hasJumpOption) {
+					extraJump.canJumpAgain = true;
+				}
+			}
+		}
+
+		/// <summary>
+		/// Toggles the ability to double jump if it's active, otherwise forbid it.
+		/// </summary>
+		/// <param name="player">Jumping player</param>
+		internal static void ToggleJumpState(Player player) {
```

Comment:
This should be called `DisableLostJumps` or something to that effect. Its sole purpose is to disable jumps when their benefactor is lost during airtime (also update the comment with that)